app/javascript/mastodon/actions/picture_in_picture.ts

```diff
@@ -0,0 +1,29 @@
+import { createAction } from '@reduxjs/toolkit';
+
+import type { PIPMediaProps } from 'mastodon/reducers/picture_in_picture';
+import { createAppAsyncThunk } from 'mastodon/store/typed_functions';
+
+interface DeployParams {
+  statusId: string;
+  accountId: string;
+  playerType: 'audio' | 'video';
+  props: PIPMediaProps;
+}
+
+export const removePictureInPicture = createAction('pip/remove');
+
+export const deployPictureInPictureAction =
+  createAction<DeployParams>('pip/deploy');
+
+export const deployPictureInPicture = createAppAsyncThunk(
+  'pip/deploy',
+  async (args: DeployParams, { dispatch, getState }) => {
+    // Do not open a player for a toot that does not exist
+
+    // @ts-expect-error state.statuses is not yet typed
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-call
+    if (getState().hasIn(['statuses', statusId])) {
```

Comment:
Shouldn't this be `args.statusId`? Isn't `statusId` undefined here? I'm surprised this passes typechecking

Reply:
Good catch. Typescript does not get it because I am using `@ts-expect-error` for this line because the Immutable state (`statuses`) is not typed yet, and it also silences this error 🤦